public/test/test-utils.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { render as rtlRender } from '@testing-library/react';
+import { AnyAction, configureStore } from '@reduxjs/toolkit';
+import { ThunkMiddlewareFor } from '@reduxjs/toolkit/dist/getDefaultMiddleware';
+import { Provider } from 'react-redux';
+import { createRootReducer } from 'app/core/reducers/root';
+import { StoreState } from 'app/types';
+import { mockNavModel } from './mocks/navModel';
+
+function render(
+  ui: React.ReactElement,
+  {
+    preloadedState,
+    store = configureStore<StoreState, AnyAction, ReadonlyArray<ThunkMiddlewareFor<StoreState, { thunk: true }>>>({
+      reducer: createRootReducer(),
+      preloadedState: {
+        navIndex: mockNavModel,
+        ...preloadedState,
+      },
+      middleware: (getDefaultMiddleware) =>
+        getDefaultMiddleware({ thunk: true, serializableCheck: false, immutableCheck: false }),
+    }),
+    ...renderOptions
+  }: { preloadedState?: Partial<StoreState>; store?: ReturnType<typeof configureStore> } = {}
+) {
+  function Wrapper({ children }: { children: React.ReactNode }) {
+    return <Provider store={store}>{children}</Provider>;
+  }
+
+  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
+}
+
+export * from '@testing-library/react';
```

Comment:
> dunno how much value we get from reexporting this. think we could just rename this render function to something like `renderWithStore`, and import `screen` as normal in the tests

same comment from: https://github.com/grafana/grafana/pull/45008#discussion_r803740065

my 2c: when i see `screen`/`fireEvent`/`waitFor` imported from `@testing-library/react` in a test, i know what that does. when it's instead imported from `test/test-utils` i can guess, but now i have to click through to make sure they're not doing anything extra that i don't need and go through some extra mental effort. 🤷 

it's more a style preference than anything so i don't mind too much if you don't wanna change it.